components/question/Textfield/index.jsx

```diff
@@ -0,0 +1,67 @@
+import PropTypes from 'prop-types';
+import { useEffect, useState } from 'react';
+
+import Input from '@/components/Input';
+
+import { buildSurveyQuestionsWithAnswers } from '../Details/buildSurveyQuestionsWithAnswers';
+import { buildTextfieldData } from './buildTextfieldData';
+
+export const textfieldTestIds = {
+  base: 'textfield',
+  input: 'textfield__input',
+};
+
+const Textfield = ({ id, survey, answers, setSurveyQuestionsWithAnswers }) => {
+  const [textFieldData, setTextFieldData] = useState([]);
+
+  useEffect(() => {
+    setTextFieldData(buildTextfieldData(survey, answers));
+  }, [id, survey, answers]);
+
+  const onInputValueChange = (value, answerId) => {
+    const surveyQuestionsWithAnswers = buildSurveyQuestionsWithAnswers(id, answerId, value);
+    const newAnswerObject = surveyQuestionsWithAnswers.answers[0];
+
+    setSurveyQuestionsWithAnswers(prevSurveyQuestionWithAnswer => {
+      const newSurveyQuestionWithAnswer = prevSurveyQuestionWithAnswer.slice();
+      const questionObject = newSurveyQuestionWithAnswer.find(response => response.id === id);
+
+      if (!questionObject) {
+        return [...newSurveyQuestionWithAnswer, surveyQuestionsWithAnswers];
+      }
+
+      let answerObject = questionObject.answers.find(answer => answer.id === answerId);
+      if (answerObject) {
+        answerObject.answer = newAnswerObject.answer;
+      } else {
+        questionObject.answers.push(newAnswerObject);
+      }
+
+      return newSurveyQuestionWithAnswer;
+    });
+  };
+
+  return (
+    <div className="mt-16" data-test-id={textfieldTestIds.base}>
+      {textFieldData.map(textField => (
+        <Input
```

Comment:
Just saw that there's an `input_mask` attributes so not sure whether we need to work on validation or not 🤔  but as there's no design for validation so I guess no 😅 
![Screen Shot 2565-05-19 at 11 16 22](https://user-images.githubusercontent.com/1772999/169203394-75544b33-52bf-49e3-82ec-97e9b4488337.png)